### PR TITLE
Add label! filter to container filters

### PR DESCRIPTION
**Pull request by xtexChooser**
#### Does this PR introduce a user-facing change?
No

```release-note
Add label! filter to containers and pods filters
```

### CODE REVIEW

**Comment by xtexChooser** [comment]
can anyone tell me if the CI server is ok?

**Comment by edsantiago** [comment]
Thank you for your PR! CI has passed -- the blue-icon "rpm-build" jobs failed, but those fail very often so we just ignore them (they are not needed for merging).

Broadly, I'm a little uncomfortable with the implementation (I would prefer to see a more general-purpose mechanism for negative tests) but I don't know how that could be done; maybe others on the team can suggest ideas.

The test could use some work: right now it's only a no-error-thrown test, not an actual functionality test. I will take a closer look at that code tomorrow, during my workday, and see if I can come up with ideas on that.

Thanks again!

**Comment by rhatdan** [comment]
Strange that we have podman pod ps --filter label!...  But not podman ps.
Code LGTM, but additional tests would be good, Also did you look at the podman pod ps implementation?

**Comment by xtexChooser** [comment]
@rhatdan no i dont and it does not
```
xtex@xtex ~> podman pod ls -f=label!=app
Error: label! is an invalid filter
xtex@xtex ~ [125]> podman pod ls -f=label=app
POD ID        NAME        STATUS      CREATED      INFRA ID      # OF CONTAINERS
6ff0ef1111a9  caddy       Exited      3 hours ago  aacac6773972  2
```
the docs is just like to be copied from `network-ls`.
I have updated the style of documentation to keep synchronized with other pages(`network-ls` and `volume-ls`, etc.)

@edsantiago I triggered a rebuild with a empty commit (:
before that, almost all builds (except rpm-builds) on CirrusCI are failed with a Java ExecutorService exception from CirrusCI

**Comment by edsantiago** [comment]
@xtexChooser could you please apply the following diff? This will actually test your added functionality. It also adds a test for `label=`, which apparently we've never had (the existing test code was pretty much a NOP).

Also, please squash your commits before re-pushing.

Thank you!

```diff
diff --git a/test/apiv2/20-containers.at b/test/apiv2/20-containers.at
index f71aae0b5..6906273fe 100644
--- a/test/apiv2/20-containers.at
+++ b/test/apiv2/20-containers.at
@@ -409,6 +409,9 @@ t GET containers/json 200 \
 podman stop bar
 
 #compat api list containers sanity checks
+podman run -d --rm --name labelcontainer_with $ENV_WORKDIR_IMG top
+podman run -d --rm --name labelcontainer_without $IMAGE top
+
 t GET containers/json?filters='garb1age}' 500 \
     .cause="invalid character 'g' looking for beginning of value"
 t GET containers/json?filters='{"label":["testl' 500 \
@@ -436,8 +439,15 @@ t POST containers/prune?filters='{"label":["tes' 500 \
 t POST libpod/containers/prune?filters='{"label":["tes' 500 \
     .cause="unexpected end of JSON input"
 
-t GET libpod/containers/json?filters='{"label":["testlabel"]}' 200 length=0
-t GET libpod/containers/json?filters='{"label!":["testlabel"]}' 200 length=0
+t GET libpod/containers/json?filters='{"label":["created_by"]}' 200 \
+  length=1 \
+  .[0].Names[0]="labelcontainer_with"
+t GET libpod/containers/json?filters='{"label!":["created_by"]}' 200 \
+  length=1 \
+  .[0].Names[0]="labelcontainer_without"
+t GET libpod/containers/json?filters='{"label!":["testlabel"]}' 200 length=2
+
+podman stop -t0 labelcontainer_with labelcontainer_without
 
 # libpod api: do not use list filters for prune
 t POST libpod/containers/prune?filters='{"name":["anyname"]}' 500 \
```

**Comment by xtexChooser** [comment]
done, thanks!

**Comment by xtexChooser** [comment]
how?

**Comment by edsantiago** [comment]
`git commit --amend -s`, and remove the extraneous cruft.

**Comment by TomSweeneyRedHat** [comment]
One doc nit, the rest LGTM
You also have trailing white space(s) somewhere, I've used `sed -i 's/\s*$//g' file_name ` in the past to make them go away.

File: docs/source/markdown/podman-ps.1.md [hunk 1]
```diff
@@ -45,20 +45,20 @@ Filters with the same key work inclusive with the only exception being
...
-| since      | [ID] or [Name] Containers created since this container                           |
-| volume     | [VolumeName] or [MountpointDestination] Volume mounted in container              |
-| health     | [Status] healthy or unhealthy                                                    |
-| pod        | [Pod] name or full or partial ID of pod                                          |
-| network    | [Network] name or full ID of network                                             |
+| **Filter** | **Description**                                                                                      |
+| ---------- | ---------------------------------------------------------------------------------------------------- |
+| id         | [ID] Container's ID (CID prefix match by default; accepts regex)                                     |
+| name       | [Name] Container's name (accepts regex)                                                              |
+| label      | [Key] or [Key=Value] Label assigned (or without, in the case of label!=[...] is used) to a container |
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply docs/source/markdown/podman-ps.1.md:52]: I would prefer to see these documented as two different filters (label, label!)
```
| label      | [Key] or [Key=Value] Label assigned  to a container |
| label!     | [Key] or [Key=Value] Label  not assigned to a container |
```

[the remark this answers is not in the document]
Comment (xtexChooser) [reply docs/source/markdown/podman-ps.1.md]: I prefer, too. should other docs be changed too? (it is out of range of this PR, so maybe someone can update other docs?

[the remark this answers is not in the document]
Comment (rhatdan) [reply docs/source/markdown/podman-ps.1.md]: I think so, since this is far clearer for Human readers.

[the remark this answers is not in the document]
Comment (TomSweeneyRedHat) [reply docs/source/markdown/podman-ps.1.md]: That's a better thought @rhatdan, I'd prefer the two lines too.

**Comment by rhatdan** [comment]
/lgtm
/hold
Thanks @xtexChooser 
If the tests pass, feel free to cancel the hold.

**Comment by rhatdan** [comment]
Filter tests are failing.

**Comment by xtexChooser** [comment]
sry. I just fixed the python tests.

also I am sure that the old test in 449cc7a5c217dbc1296f2f3fcc63d0b9b9826bec is broken as `=` is URL-encoded.

**Pull request merged**